Frameworks/Foundation/NSString.mm

```diff
@@ -2021,7 +2021,13 @@ - (id)initWithCoder:(NSCoder*)coder {
  @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)coder {
-    [coder encodeObject:self forKey:@"NS.string"];
+    if ([coder isKindOfClass:[NSKeyedArchiver class]]) {
```

Comment:
I am not happy about having to do this.
However.

Without this escape hatch, our `NSKeyedArchiver` will encode `[NSMutableString stringWithString:@"x"]` as:

```
0: $null
1: {
    class: <2>
    NS.string: <1>
}
2: {
    $className: NSMutableString
    ...
}
```

(where the string is object **1** and `<x>` indicates a reference to object *x*)